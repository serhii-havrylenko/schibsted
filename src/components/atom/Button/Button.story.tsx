import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react';
import * as React from 'react';

import Button from './Button';

storiesOf('Button', module).add('default', () => (
  <Button onClick={action('clicked')} />
));
