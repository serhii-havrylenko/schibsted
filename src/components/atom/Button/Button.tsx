/**
 * Provides basic button
 * @module #atom/Button
 */
import Btn from '@material-ui/core/Button';
import * as React from 'react';

export interface Button {
  onClick?: () => void;
}

/**
 * Provides basic button component
 */
const Button: React.SFC<Button> = ({ children, onClick }) => (
  <Btn color="primary" onClick={onClick}>
    {children || 'sample'}
  </Btn>
);

export default Button;
