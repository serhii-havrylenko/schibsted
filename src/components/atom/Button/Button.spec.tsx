import { mount } from 'enzyme';
import * as React from 'react';

import Button from './Button';

jest.mock('@material-ui/core/Button');

describe('<Button /> tests', () => {
  test('should match snapshot', () => {
    const tree = mount(<Button onClick={undefined} />);
    expect(tree).toMatchSnapshot();
  });
});
