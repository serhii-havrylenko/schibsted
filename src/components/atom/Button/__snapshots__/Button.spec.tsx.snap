// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> tests should match snapshot 1`] = `
<Button>
  <WithStyles(Button)
    color="primary"
  />
</Button>
`;
